pyccel/parser/syntactic.py

```diff
@@ -439,51 +434,60 @@ def _visit_UnitaryOperatorNode(self, stmt):
 
     def _visit_BinaryOperatorNode(self, stmt):
 
-        first = self._visit(stmt.first)
+        first  = self._visit(stmt.first)
         second = self._visit(stmt.second)
+        assoc  = isinstance(stmt.second, AssociativeParenthesisNode)
+        cond   = isinstance(second, (PyccelMul, PyccelDiv, PyccelFloorDiv, PyccelMod))
+        cond   = cond and not assoc
+
         if stmt.value == '+':
-            return Add(first, second)
-        elif stmt.value == '*':
+            return PyccelAdd(first, second)
 
+        elif stmt.value == '*':
             if isinstance(first, (PythonTuple, Tuple, List)):
                 return Dlist(first, second)
-            return Mul(first, second)
+            if cond:
+                args  = second.args
+                first = PyccelMul(first, args[0])
+                return second.func(first, *args[1:])
```

Comment:
I'm not clear on what you are doing here to ensure that the brackets are created. Have you tested this thouroughly?

Reply:
it's been changed , and yes I have added the tests in `pyccel/scrtipy/expressions.py`